src/components/GetRooms.vue

```diff
@@ -36,30 +36,35 @@ const config = {
 const app = firebase.initializeApp(config);
 const auth = firebase.auth();
 const db = getFirestore(app);
+const roomsCol = collection(db, "rooms");
 
-async function getRooms(db) {
-  const roomsCol = collection(db, "rooms");
+async function getRooms() {
   const roomsSnapshot = await getDocs(roomsCol);
   const roomsList = roomsSnapshot.docs;
   // const roomsList = roomsSnapshot.docs.map((doc) => doc.data());
   return roomsList;
 }
 
+function subscribeToRoomUpdates(onUpdate){
```

Comment:
I expect this might be tricky to get your head around.

`subscribeToRoomsUpdates` expects to be passed a function.

`subscribeToRoomsUpdates` will "call back" that function with the rooms. But those rooms need to be the same "shape" as what we get back from `getRooms` because that's what our template needs. So that's why L50 passes `rooms.docs` to `onUpdate`.

Reply:
Yeah it's a bit confusing. `onUpdate` seems to be born ex nihilo. I can't see it defined anywhere or pulled in from the imports above

I can see how it's an argument so we can kind of just name it, but then it seems to be doing work inside the function.

Also just struggling as usual with reading arrow functions, especially when they are nested inside other functions. Maybe I just need to drill that.